daft/catalog/__unity.py

```diff
@@ -94,12 +95,16 @@ def _try_from(obj: object) -> UnityTable | None:
             return UnityTable(obj)
         return None
 
-    @property
-    def inner(self) -> InnerTable:
-        """Returns the inner unity table."""
-        return self._inner
+    ###
+    # read methods
+    ###
+
+    def read(self, **options) -> DataFrame:
+        return read_deltalake(self._inner, version=options.get("version"))
 
-    def read(self) -> DataFrame:
-        import daft
+    ###
+    # write methods
+    ###
 
-        return daft.read_deltalake(self._inner)
+    def write(self, df: DataFrame | object, mode: str = "append", **options):
```

Comment:
can we make `mode` a literal of the different write modes instead of just `str`. 

`mode: Literal["append", "overwrite", ...] = "append"`